src/Plugin/HistoryPlugin.php

```diff
@@ -40,8 +40,10 @@ public function handleRequest(RequestInterface $request, callable $next, callabl
             $journal->addSuccess($request, $response);
 
             return $response;
-        }, function (Exception $exception) use ($request, $journal) {
-            $journal->addFailure($request, $exception);
+        }, function (\Exception $exception) use ($request, $journal) {
+            if ($exception instanceof Exception) {
+                $journal->addFailure($request, $exception);
```

Comment:
Other exceptions should be handled by your code or logged/"handled" by the framework (end up in 500 usually). This Journal is to record HTTP messages and potential failures of them. This is the same questions as domain/http related exceptions.